### PR TITLE
Python3 upgrade

### PR DESCRIPTION
**Pull request by DavidCDreher**
This PR makes flask-genshi Python2&3 compatible as well as incorporate a fix for jinja filters from https://github.com/vthriller/flask-genshi

The changes follow closely https://github.com/vthriller/flask-genshi but with a cleaner commit history and without dropping flatland support since it is still maintained.

Change list: 
- Make compatible with `werkzeug>1.0`
- Drop deprecated Flask module support
- Migrate from abandoned `attest` framework to `pytest`
- Migrate from deprecated flask extension syntax `flaskext.genshi` to modern `flask_genshi` naming

Attached is the json output of a tox run for python versions: `py27, py36, py37, py38`.

[flask_genshi_tox.zip](https://github.com/dag/flask-genshi/files/5169258/flask_genshi_tox.zip)

### CODE REVIEW

File: docs/flask.rst [hunk 1]
```diff
@@ -37,21 +37,6 @@ Result:
 .. versionchanged:: 0.6 Top-level access.
 
 
-Module templates
```
Comment (ReeceStevens) [inline docs/flask.rst:40]: Thank you for removing stale documentation as you go through the migration process!

File: tests/test_context_processors.py [hunk 2]
```diff
@@ -13,12 +14,12 @@ def inject_rudolf():
...
         rendered = render_response("context.html")
 
-        expected_data = inspect.cleandoc(
+        expected_data = cleandoc(
             """
             <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
             <pre>rudolf = The red-nosed reindeer</pre>
             """
-        )
+        ).encode("UTF-8")
```
Comment (ReeceStevens) [inline tests/test_context_processors.py:23]: Might be worth leaving a small comment here that the encoding is required due to `render_response` returning bytes

**Review by ReeceStevens — APPROVED** [review]
@DavidCDreher this PR looks great, I don't really have any comments. I wasn't super clear on some of the jinja environment workarounds but there's enough context to put the pieces together. Excellent commits, too!

File: tests/test_context_processors.py (hunk 2 above)
Comment (DavidCDreher) [reply tests/test_context_processors.py:23]: Updated with comments!

**Pull request merged**